File: src/test/scala/com/imcode/imcms/test/config/EnvironmentConfig.scala
package com.imcode
package imcms.test.config

import org.springframework.context.annotation._
import javax.inject.Inject
import org.springframework.core.env.{StandardEnvironment, Environment}

@Configuration
@PropertySource(value = Array("classpath:server.properties"))
class EnvironmentConfig {

  var env: Environment = _

  @Inject
  def setEnv(env: StandardEnvironment) {
    env.getPropertySources.remove(StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME)
    this.env = env
  }
}

File: src/test/java/com/imcode/imcms/test/EnvironmentConfig.java
package com.imcode.imcms.test;

import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;
import org.springframework.core.env.StandardEnvironment;

import javax.inject.Inject;

@Configuration
@PropertySource("classpath:server.properties")
public class EnvironmentConfig {

    @Inject
    public void setEnv(StandardEnvironment env) {
        env.getPropertySources().remove(StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME);
    }
}
